gpytorch/models/approximate_gp.py

```diff
@@ -76,6 +76,30 @@ def pyro_model(self, input, beta=1.0, name_prefix=""):
         return super().pyro_model(input, beta=beta, name_prefix=name_prefix)
 
     def get_fantasy_model(self, inputs, targets, **kwargs):
```

Comment:
I can see why it might be convenient to always have `get_fantasy_model` return an `ExactGP`, regardless of the original model class, but it might be worth considering naming this something else, reserving `get_fantasy_model` for the version of OVC that returns a variational GP (in other words consider requiring that `get_fantasy_model` always returns an instance of the original class).

Reply:
That was a thought I originally had but it requires the reasonably unstable updates to `m`, `S` in order to return its own model class itself rather than the exactGP. Although potentially lower overhead in the future to implement new fantasization strategies.